packages/ember-data/lib/system/model/states.js

```diff
@@ -658,6 +658,11 @@ var RootState = {
       becameError: function(record) {
         record.transitionTo('uncommitted');
         record.triggerLater('becameError', record);
+      },
+
+      becameInvalid: function(record) {
+        record.transitionTo('invalid');
+        record.triggerLater('becameInvalid', record);
```

Comment:
I thought, it needed to be called because `DirtyState.inFlight.becameInvalid` calls `record.send('invokeLifecycleCallbacks')` and then:
```js
invalid: {
    invokeLifecycleCallbacks: function(record) {
      record.triggerLater('becameInvalid', record);
    }
}
```

The implementation is based on other `invalid event` implementations but honestly, I didn't understand well when `record.triggerLater('becameInvalid', record);` is necessary.

If I removed this `triggerLater`, the test pass. Do you want to me to update PR?